### PR TITLE
[8.x] Add mergeWhen method to request

### PR DESCRIPTION
**Pull request by lakuapik**
laravel [v8.55.0](https://github.com/laravel/framework/releases/tag/v8.55.0) introduce support for conditional validation.  
basically this PR has same goal but for the request class.  
this will add `mergeWhen` method to the request class.

Q: why?  
A: it might be useful when we need a simple conditional logic to merge input to the request

consider this following scenario:

```php
// its a `create new task form` which has team_id field
// when user does not supply the team_id, the default should be their own team 

// before
public function store(Request $request) 
{
    if ($request->isNotFilled('team_id')) {
        $request->merge([
            'team_id' => Auth::user()->team_id,
            'foo' => 'bar', // anything...
        ]);
    }
    $request->validate([
        'team_id' => 'required|exists:teams,id',
        //...
    ]);
    //...
}

// after
public function store(Request $request) 
{
    $request->mergeWhen($request->isNotFilled('team_id'), [
        'team_id' => Auth::user()->team_id,
        'foo' => 'bar', // anything...
    ]);
    $request->validate([
        'team_id' => 'required|exists:teams,id',
        //...
    ]);
    //...
}
//...
```

this will save `4 spaces of identation` 😁️
feel free to accept or drop this PR 😇️

### CODE REVIEW

**Comment by lakuapik** [comment]
^force-pushed to fix styleci.

**Comment by morloderex** [comment]
@lakuapik I believe you didn't had to to that sense styleci would be applied upon merge if i remember correctly.

File: src/Illuminate/Http/Request.php [hunk 1]
```diff
@@ -331,6 +331,22 @@ public function merge(array $input)
...
+    /**
+     * Conditionally merge new input into the current request's input array.
+     *
+     * @param  callable|bool  $condition
+     * @param  array  $input
+     * @return $this
+     */
+    public function mergeWhen($condition, array $input)
+    {
+        $shouldMerge = is_callable($condition) ? call_user_func($condition) : $condition;
```
Comment (morloderex) [inline src/Illuminate/Http/Request.php:343]: There might be an oppitunity to use the value helper method here?

**Comment by bert-w** [comment]
You could just do:
``` php
$request->merge($request->isNotFilled('team_id') ? [
    'team_id' => Auth::user()->team_id,
    'foo' => 'bar', // anything...
] : []);
```

Not the prettiest thing I know (and it may try to merge an empty array).

**Pull request closed without merging**